cg/apps/sequencing_metrics_parser/parsers/bcl_convert.py

```diff
@@ -106,15 +107,16 @@ def get_sample_internal_ids(self) -> List[str]:
         """Return a list of sample internal ids."""
         sample_internal_ids: List[str] = []
         for sample_demux_metric in self.demux_metrics:
-            if self.is_valid_sample_project(sample_project=sample_demux_metric.sample_project):
+            if self.is_valid_sample(sample_internal_id=sample_demux_metric.sample_internal_id):
                 sample_internal_ids.append(sample_demux_metric.sample_internal_id)
         return list(set(sample_internal_ids))
 
-    def is_valid_sample_project(self, sample_project: str) -> bool:
+    def is_valid_sample(self, sample_internal_id: str) -> bool:
         """Return True if the sample project is valid."""
-        if sample_project in [INDEX_CHECK, UNDETERMINED]:
-            return False
-        return True
+        pattern = r"^[A-Za-z]{3}[0-9]{3}.*"
+        if re.match(pattern, sample_internal_id) is not None:
+            return True
+        return False
```

Comment:
These changes are unintended. It seems like they were not stashed properly when switching branches. Nevertheless, this part of the code is working now as well. Furthermore, this code will be refactored once the current PR with the sample sheet functionality is merged to make use of that functionality.

Reply:
It looks good though